### PR TITLE
Get version info for remote istiod and refactor status

### PR DESCRIPTION
**Pull request by nrfox**
### Describe the change

Gets the version for a remote istiod by port fowarding to the istiod pod and doing a `/version` request to the http monitoring port (15014).

Updates the default for `config.ExternalServices.Istio.UrlServiceVersion` to be `""`. If it is set to empty string, it will be autodetected and no longer needs to be set manually especially with multiple revisions. If it is set, the value will be used on the home cluster.

Also refactors some things out of `status/status` since there were circular imports with the `business` package.

### Steps to test the PR

1. Setup local multi-primary with kind
```
hack/run-integration-tests.sh --test-suite frontend-multi-primary --setup-only true
```
2. Edit an istiod deployment's image on the "west" cluster down to n-1 e.g. from `1.22.1` to `1.21.0`
3. Visit the mesh page and see the different version on the remote cluster: 
Screenshot from 2024-06-10 11-00-12

### Automation testing

None created.

### Issue reference

Relates to #7177 and https://github.com/kiali/kiali/issues/7203

### CODE REVIEW

**Review by jshaughn — APPROVED** [review]
A couple of minor comments, but nothing blocking.  Sanity test worked. LG!

File: mesh/config/cytoscape/cytoscape.go [hunk 1]
```diff
@@ -129,13 +129,12 @@ func NewConfig(meshMap mesh.MeshMap, o mesh.ConfigOptions) (result Config) {
 	})
 
 	elements := Elements{nodes, edges}
-	meshName := mesh.StatusGetter().Status["Mesh Version"]
-	if meshName == "" {
-		meshName = "Istio Mesh"
+	if o.MeshName == "" {
+		o.MeshName = "Unknown"
```
[the remark this answers is not in the document]
Comment (nrfox) [reply mesh/config/cytoscape/cytoscape.go:133]: If there's a default that istio assigns we should use that.

File: mesh/generator/generator.go [hunk 2]
```diff
@@ -91,12 +94,6 @@ func BuildMeshMap(ctx context.Context, o mesh.Options, gi *mesh.AppenderGlobalIn
 
 		name := cp.IstiodName
 
-		version := esVersions["Istio"]
-		if isCanary {
-			// for canary upgrade the version is the revision name with '.' instead of '-'
-			version = strings.Replace(cp.Revision, "-", ".", -1)
-		}
```
[the remark this answers is not in the document]
Comment (nrfox) [reply mesh/generator/generator.go:98]: Yes and I think technically revision could be anything like `production` or `development`. Doesn't necessarily have to include version at all.

File: mesh/config/cytoscape/cytoscape.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jshaughn) [reply mesh/config/cytoscape/cytoscape.go]: I don't think there is.  Maybe we should just leave it unset and deal with "" as needed in the UI.  Not a big deal, not a blocker.

**Review by jshaughn — APPROVED** [review]

**Pull request merged**